### PR TITLE
MM-40918-Highlighted Enterprise Features after Starting an Enterprise…

### PR DESCRIPTION
**Pull request by julmondragon**
#### Summary
Highlighted Enterprise Features after Starting an Enterprise Trial

- Update text and images for trial benefits modal for top 3 features:
  - Guest accounts
  - Mobile > Secure ID only push notifications
  - Compliance > Automated compliance exports
- Add Telemetry for:
  - Post Enterprise Modal Page number views
  - Modal next / prev click
- Enhancement for TrialBenefitModal component
  - Component structure
  - Styling for consistency between Admin console and App contexts

#### Ticket Link
[MM-40918](https://mattermost.atlassian.net/browse/MM-40918)

#### Screenshots

Trial start
trial_start_benefits_modal

Post trial
post_trial_benefits_modal

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: components/common/carousel/carousel.tsx [hunk 1]
```diff
@@ -25,30 +25,30 @@ const Carousel: React.FC<Props> = ({
...
 
     const prevSlide = () => {
         setNextButtonDisabled(false);
-        if (slideIndex !== 1) {
-            setSlideIndex(slideIndex - 1);
-        } else if (slideIndex === 1) {
-            if (infiniteSlide) {
-                setSlideIndex(dataSlides.length);
+
+        const newSlideIndex = slideIndex === 1 && infiniteSlide ? dataSlides.length : (slideIndex !== 1 && slideIndex - 1) || undefined;
```
Comment (pablovelezvidal) [inline components/common/carousel/carousel.tsx:43]: nit: this kind of tertiary conditionals are hard to read, what if you self-document the code by storing in a variable with a meaningful name each of the options, like `const x = theCondition ? meaningfulName1 : meaningfulName2;`

**Review by pablovelezvidal — APPROVED** [review]
LGTM @julmondragon

File: components/common/carousel/carousel.tsx (hunk 1 above)
Comment (julmondragon) [reply components/common/carousel/carousel.tsx]: Updated!

**Comment by julmondragon** [comment]
> 

Hi @AGMETEOR , this will be covered in https://mattermost.atlassian.net/browse/MM-40286 as part of an ongoing effort to add e2e tests for cloud features (https://mattermost.atlassian.net/browse/MM-30845).

File: components/common/carousel/carousel.tsx (hunk 1 above)
Comment (pablovelezvidal) [reply components/common/carousel/carousel.tsx]: excelent, thanks!

**Comment by saturninoabril** [comment]
(Sorry for chiming in)

@julmondragon For new feature, I would recommend to add E2E in the PR itself to protect from future regression and not to add up to the list of manual tests for release testing. A good starting point before writing E2E is to have a test case saved into Zephyr. Let Steve know if you need help with Zephyr and Dev/SDETs when writing Cypress.

**Comment by marianunez** [comment]
> (Sorry for chiming in)
> 
> @julmondragon For new feature, I would recommend to add E2E in the PR itself to protect from future regression and not to add up to the list of manual tests for release testing. A good starting point before writing E2E is to have a test case saved into Zephyr. Let Steve know if you need help with Zephyr and Dev/SDETs when writing Cypress.

@saturninoabril this is actually mostly a texts and images changes of an existing functionality. Agreed that we need to add E2E tests for all PRs but in this case in particular we need to look at it a bit more given that it involves downloading a trial license from CWS and I am not sure we have the appropriate setup for this yet.

**Review by stevemudie — APPROVED** [review]
Looks good @julmondragon!

**Pull request merged**